File: server/src/main/java/com/logiflow/server/models/ChatMessage.java
package com.logiflow.server.models;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "chat_messages", indexes = {
        @Index(name = "idx_chat_trip_id_created_at", columnList = "trip_id, created_at")
})
public class ChatMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "message_id")
    private Long messageId;

    @Column(name = "trip_id", nullable = false)
    private Integer tripId;

    @Column(name = "sender_username", nullable = false, length = 100)
    private String senderUsername;

    @Column(name = "sender_role", nullable = true, length = 50)
    private String senderRole;

    @Column(name = "recipient_driver_id", nullable = true)
    private Integer recipientDriverId;

    @Column(name = "content", nullable = false, length = 2000)
    private String content;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: client_mobile/lib/models/chat/chat_message.dart
class ChatMessage {
  final int? tripId;
  final String senderUsername;
  final String? senderRole;
  final int? recipientDriverId;
  final String content;
  final DateTime createdAt;
  final int? messageId;

  ChatMessage({
    this.tripId,
    required this.senderUsername,
    this.senderRole,
    this.recipientDriverId,
    required this.content,
    required this.createdAt,
    this.messageId,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    final createdAtRaw = json['createdAt'];
    return ChatMessage(
      messageId: json['messageId'] as int?,
      tripId: json['tripId'] as int?,
      senderUsername: json['senderUsername'] as String? ?? 'Unknown',
      senderRole: json['senderRole'] as String?,
      recipientDriverId: json['recipientDriverId'] as int?,
      content: json['content'] as String? ?? '',
      createdAt: createdAtRaw is String
          ? (DateTime.tryParse(createdAtRaw) ?? DateTime.now())
          : DateTime.now(),
    );
  }
}
